vm/src/stdlib/syslog.rs

```diff
@@ -0,0 +1,181 @@
+    #[pyattr]
+    const LOG_MAIL: i32 = libc::LOG_MAIL;
+    #[pyattr]
+    const LOG_NDELAY: i32 = libc::LOG_NDELAY;
+    #[pyattr]
+    const LOG_NEWS: i32 = libc::LOG_NEWS;
+    #[pyattr]
+    const LOG_NOTICE: i32 = libc::LOG_NOTICE;
+    #[pyattr]
+    const LOG_NOWAIT: i32 = libc::LOG_NOWAIT;
+    #[pyattr]
+    const LOG_ODELAY: i32 = libc::LOG_ODELAY;
+    #[cfg(not(target_os = "redox"))]
+    #[pyattr]
+    const LOG_PERROR: i32 = libc::LOG_PERROR;
+    #[pyattr]
+    const LOG_PID: i32 = libc::LOG_PID;
+    #[pyattr]
+    const LOG_SYSLOG: i32 = libc::LOG_SYSLOG;
+    #[pyattr]
+    const LOG_USER: i32 = libc::LOG_USER;
+    #[pyattr]
+    const LOG_UUCP: i32 = libc::LOG_UUCP;
+    #[pyattr]
+    const LOG_WARNING: i32 = libc::LOG_WARNING;
+
+    fn get_argv(vm: &VirtualMachine) -> PyResult {
+        let argv = sysmodule::argv(vm);
+        let scriptobj = PyStrRef::try_from_object(vm, argv.get_item(0.into_pyobject(vm), vm)?)?;
+        let scriptobj = scriptobj.as_ref();
+
+        let scriptlen = scriptobj.len();
+        if scriptlen == 0 {
+            return Err(vm.new_value_error("empty script was given".to_owned()));
+        }
+
+        let retstr = match scriptobj.find('\\') {
+            Some(value) => scriptobj[value..scriptlen].to_string(),
+            None => scriptobj.to_string(),
+        };
+
+        Ok(retstr.into_pyobject(vm))
+    }
+
+    fn convert_cstring(s: &str, vm: &VirtualMachine) -> PyResult<std::ffi::CString> {
+        std::ffi::CString::new(s).map_err(|err| err.into_pyexception(vm))
+    }
+
+    #[derive(FromArgs)]
+    struct OpenLogArgs {
+        #[pyarg(any, optional)]
+        ident: OptionalOption<PyStrRef>,
+        #[pyarg(any, optional)]
+        logoption: OptionalArg<i32>,
+        #[pyarg(any, optional)]
+        facility: OptionalArg<i32>,
+    }
+
+    #[pyfunction]
+    fn openlog(args: OpenLogArgs, vm: &VirtualMachine) -> PyResult<()> {
+        let logoption = args.logoption.unwrap_or(0);
+        let facility = args.facility.unwrap_or(LOG_USER);
+        let ident = match args.ident.flatten() {
+            Some(s) => s,
+            None => {
+                let argv = get_argv(vm)?;
+                PyStrRef::try_from_object(vm, argv)?
+            }
+        };
+
+        let cident = convert_cstring(ident.as_ref(), vm)?;
+        unsafe { libc::openlog(cident.as_ptr(), logoption, facility) };
+        Ok(())
+    }
+
+    #[derive(FromArgs)]
+    struct SysLogArgs {
+        #[pyarg(positional)]
+        priority: PyObjectRef,
+        #[pyarg(positional, optional)]
+        message_object: OptionalOption<PyStrRef>,
+    }
+
+    #[pyfunction]
+    fn syslog(args: SysLogArgs, vm: &VirtualMachine) -> PyResult<()> {
+        let (priority, msg) = match args.message_object.flatten() {
+            Some(s) => (
+                PyIntRef::try_from_object(vm, args.priority)?
+                    .as_bigint()
+                    .to_i32()
+                    .unwrap(),
```

Comment:
This is a user input, so I think user can input anything out of range value here.

```suggestion
                i32::try_from_object(vm, args.priority)?,
```